include/aws/http/request_response.h

```diff
@@ -497,7 +501,74 @@ AWS_HTTP_API
 void aws_http_headers_clear(struct aws_http_headers *headers);
 
 /**
- * Create a new request message.
+ * Get the `:method` value (HTTP/2 headers only).
+ */
+AWS_HTTP_API
+int aws_h2_headers_get_request_method(const struct aws_http_headers *h2_headers, struct aws_byte_cursor *out_method);
```

Comment:
if these are part of the public API, use "aws_http2"
we only use "h2" in private